### ISSUE TITLE
Error starting Selenium Grid - java.io.UncheckedIOException

### ISSUE THREAD

**Issue by msscribner** [SeleniumHQ/docker-selenium#1364]
## 🐛 Bug Report

Start the Selenium Grid, but get a java.io.UncheckedIOException.  This appears to prevent the Nodes from connecting to the Hub.

## To Reproduce
Start the Stack:  sudo docker stack deploy --compose-file ~/Desktop/docker-compose-Linux8.yml stackdemo

[docker-compose-Linux8.txt](https://github.com/SeleniumHQ/docker-selenium/files/6956435/docker-compose-Linux8.txt)

## Expected behavior
Expect to browse to Selenium Grid and view the connected nodes:   http://172.16.51.32:4444/ui/index.html#/

##Errors from Selenium-Hub log:
. . . 
2021-08-09T17:24:24.339401326Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 17:24:24.332 WARN 
[UnboundZmqEventBus$PollingRunnable.lambda$notifyListeners$2] - Caught exception from listener: org.openqa.selenium.events.EventListener@5dd7b6b8
2021-08-09T17:24:24.339515052Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | java.io.UncheckedIOException: java.net.ConnectException: Connection refused: 883b9bbcd39a/10.0.9.11:5555
2021-08-09T17:24:24.339637345Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.remote.http.netty.NettyHttpHandler.makeCall(NettyHttpHandler.java:80)
2021-08-09T17:24:24.339678535Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)
2021-08-09T17:24:24.339880056Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)
2021-08-09T17:24:24.339914467Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.remote.http.netty.NettyHttpHandler.execute(NettyHttpHandler.java:51)
2021-08-09T17:24:24.340000244Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)
2021-08-09T17:24:24.340026358Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)
2021-08-09T17:24:24.340075507Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.remote.http.netty.NettyClient.execute(NettyClient.java:116)
2021-08-09T17:24:24.340114407Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.remote.tracing.TracedHttpClient.execute(TracedHttpClient.java:55)
2021-08-09T17:24:24.341189154Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.grid.node.remote.RemoteNode.getStatus(RemoteNode.java:207)
2021-08-09T17:24:24.341207520Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.grid.distributor.local.LocalDistributor.add(LocalDistributor.java:254)
2021-08-09T17:24:24.341238260Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.grid.distributor.local.LocalDistributor.register(LocalDistributor.java:243)
2021-08-09T17:24:24.341244070Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.grid.distributor.local.LocalDistributor.lambda$new$1(LocalDistributor.java:167)
2021-08-09T17:24:24.341249082Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.events.EventListener.accept(EventListener.java:43)
2021-08-09T17:24:24.341279059Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.events.EventListener.accept(EventListener.java:25)
2021-08-09T17:24:24.341284599Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at org.openqa.selenium.events.zeromq.UnboundZmqEventBus$PollingRunnable.lambda$notifyListeners$2(UnboundZmqEventBus.java:273)
2021-08-09T17:24:24.341289492Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
2021-08-09T17:24:24.341294420Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2021-08-09T17:24:24.341299249Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2021-08-09T17:24:24.341303804Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2021-08-09T17:24:24.341308338Z stackdemo_selenium-hub.1.o5hjm5nna2sh@qadockermgr1    | 	at java.base/java.lang.Thread.run(Thread.java:829)
. . .

[hub.txt](https://github.com/SeleniumHQ/docker-selenium/files/6956451/hub.txt)

## Environment

OS: 
Docker-Selenium image version: 
Docker version:

**Comment by msscribner** [comment]
If Chrome service is set to replicate 40 times, then this issue always occurs.  
If Chrome service is set to replicate 1 time, then this issue occurs much less frequently.  

And once the exception occurs on the HUB, the Chrome service(s) attempt to register (Sending registration event...) but eventually stops... Chrome Nodes are never registered on the GRID. 

Should be noted... the Selenium-Hub Service is running on the Manager node.   I haven't been able to reproduce the error if the Selenium-Hub Service is running on a Worker node with a Chrome Service.   

If the Selenium-Hub is configured to run w/o the Chrome service...then the exception **always** occurs.

**Comment by msscribner** [comment]
I am also able to reproduce trying to Scale the Chrome Service.

- Two Node Stack (1 manager; 2 worker)
- Start the Selenium Grid 
        (Constrain HUB service to Manager node;  Constrain Chrome service to Worker node with replicas set to 0)
- Allow Selenium Grid to start
- Scale the Chrome service to 1.  Service scales 
- Scale the Chrome service to 2.  Receive the java.io.UncheckedIOException

---------------------------------------------------------------------------------------------
```
version: '3'

services:
  chrome:
    image: selenium/node-chrome:4.0.0-rc-1-prerelease-20210804
    environment:
      - SE_EVENT_BUS_HOST=selenium-hub
      - SE_EVENT_BUS_PUBLISH_PORT=4442
      - SE_EVENT_BUS_SUBSCRIBE_PORT=4443
    deploy:
      replicas: 0
      placement:
       constraints: 
        - node.role == worker
    entrypoint: bash -c 'SE_OPTS="--host $$HOSTNAME" /opt/bin/entry_point.sh'
```


  ```
selenium-hub:
    image: selenium/hub:4.0.0-rc-1-prerelease-20210804
    shm_size: 2gb
    ports:
      - "4442:4442"
      - "4443:4443"
      - "4444:4444"
    deploy:
      replicas: 1
      placement:
       constraints: 
        - node.role == manager
```
---------------------------------------------------------------------------------------------

**Errors from Selenium-Hub log:**

[hub2.txt](https://github.com/SeleniumHQ/docker-selenium/files/6992173/hub2.txt)

**Comment by msscribner** [comment]
Reconfigured Selenium Grid (HUB and Chrome) to use images: 3.141.59.20210804.   I was able to constrain the HUB to the Manager Node.  GRID started up no problem.  I was able to Scale to 50 Chrome services all constrained to Worker Nodes.   So with this configuration the HUB was able to run on a node that did not contain the Chrome services....that is the  java.io.UncheckedIOException exception does not occur.
